File: 055-字符流中第一个不重复的字符/055-FirstAppearingOnce.py
# -*- coding:utf-8 -*-
class Solution:
    # 返回对应char
    def __init__(self):
        self.count=[]
    def FirstAppearingOnce(self):
        # write code here
        return self.count[0] if self.count else '#'
    def Insert(self, char):
        # write code here
        if char not in self.count:
            self.count.append(char)
        else:
            self.count.remove(char)

File: 055-字符流中第一个不重复的字符/055-FirstAppearingOnce.cpp
class Solution
{
    string str = "";
    int count[256];
public:
    void Insert(char ch)
    {
        str += ch;
        count[(int)ch]++;
    }

    char FirstAppearingOnce()
    {
        int len = str.size( );
        for(int i = 0; i < len; i++)
        {
            if(count[(int)str[i]] == 1)
            {
                return str[i];
            }
        }
        return '#';
    }

};
